### PR TITLE
Added a wrapper for value and default value in date picker and time picker

### LINKED ISSUE #785: Default value for DatePicker breaks the application while trying to open the calendar.
Getting an error while trying to open the calender in DatePicker with a default value

```
index.js:391 Uncaught TypeError: t.weekday is not a function
    at Object.getWeekDay (index.js:391)
    at index.js:98
    at me (index.js:98)
    at renderWithHooks (react-dom.development.js:14985)
    at mountIndeterminateComponent (react-dom.development.js:17811)
    at beginWork (react-dom.development.js:19049)
    at HTMLUnknownElement.callCallback (react-dom.development.js:3945)
    at HTMLUnknownElement.func.___hb (honeybadger.js:1477)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:3994)
    at invokeGuardedCallback (react-dom.development.js:4056)
```

Code

```js
<DatePicker
   className="w-full"
   type="range" 
   defaultValue={[dayjs().subtract(1, "day"), dayjs()]}
/>
```

Clearing the value before opening the calendar works.

### PR DESCRIPTION
**Pull request by amaldinesh7**
Fixes #785 

@ajmaln _a Please review

### CODE REVIEW

File: lib/components/DatePicker.js [hunk 1]
```diff
@@ -23,28 +28,39 @@ const datePickerTypes = {
...
   const Component = datePickerTypes[type?.toLowerCase()];
   const format = showTime ? `${dateFormat} ${timeFormat}` : dateFormat;
 
   const id = useId(otherProps.id);
   const errorId = `error_${id}`;
 
+  const convertToDayjsObjects = (value) => {
+    return value instanceof Array
+      ? value.map((date) => (date ? dayjs(date) : date))
+      : value && dayjs;
```
Comment (ajmaln) [inline lib/components/DatePicker.js:61]: @amaldinesh7 _a shouldn't this be `value && dayjs(value)`?

File: lib/components/TimePicker.js [hunk 2]
```diff
@@ -50,6 +54,12 @@ const TimePicker = (
   const id = useId(otherProps.id);
   const errorId = `error_${id}`;
 
+  const convertToDayjsObjects = (value) => {
+    return value instanceof Array
+      ? value.map((date) => (date ? dayjs(date) : date))
+      : value && dayjs;
+  };
```
Comment (ajmaln) [inline lib/components/TimePicker.js:61]: Can we move this to `utils` and import and reuse it.

**Comment by amaldinesh7** [comment]
@ajmaln _a Fixed the comments. Please check.

**Review by ajmaln — APPROVED** [review]
LGTM!

**Pull request merged**